otter/integration/tests/test_convergence.py

```diff
@@ -853,6 +857,71 @@ def test_convergence_fixes_errored_building_servers(self):
         d.addCallback(lambda _: group)
         return d
 
+    @tag("CATC-010")
+    @inlineCallbacks
+    def test_servers_that_build_for_too_long_time_out_and_are_replaced(self):
+        """
+        CATC-010
+
+        Requires Mimic for error injection.
+
+        1. Mimic should cause a single server to remain in building too
+           long.
+        2. Create group with 2 servers.  (One of them will time out building.)
+        3. Check with Nova that 2 servers are built - wait for one to be
+           active.  The other is the one that should remain in build.
+        4. Wait for autoscale to show 2 servers being active.
+        5. Check with Nova to ensure that there are only 2 active servers on
+           the account.  The one that was building forever should be deleted.
+        """
+        group = self.helper.create_group(
+            image_ref=image_ref, flavor_ref=flavor_ref,
+            min_entities=2, max_entities=10,
+            server_name="build-timeout"
+        )
+
+        yield MimicNova(pool=self.helper.pool).sequenced_behaviors(
+            self.rcs,
+            criteria=[{"server_name": "build-timeout.*"}],
+            behaviors=[
+                {"name": "build",
+                 "parameters": {"duration": otter_build_timeout * 2}},
+                {"name": "default"}
+            ])
+        yield group.start(self.rcs, self)
+
+        initial_servers = yield wait_for_servers(
+            self.rcs, pool=self.helper.pool, group=group,
+            timeout=otter_build_timeout,
+            matcher=MatchesSetwise(
+                ContainsDict({'status': Equals('ACTIVE')}),
+                ContainsDict({'status': Equals('BUILD')}),
+            ))
+        building_server_id = [s['id'] for s in initial_servers
+                              if s['status'] == 'BUILD'][0]
```

Comment:
Non-blocking nitpick: I've found using `next` handy: `next(s['id'] for s in initial_servers if s['status'] == 'BUILD')`